docs/src/languages/php/tuning.md

```diff
@@ -3,216 +3,192 @@ title: "Performance tuning"
-you need to clear the OPcache explicitly on deployment (which can be done by restarting PHP-FPM).
+Determining the optimal limit to memory consumption requires executing code via a web request to get adequate statistics.
+[`CacheTool`](https://github.com/gordalina/cachetool) is an open-source tool to help you get the statistics.
+
+To determine the total amount of memory to use:
+
+1. Connect to the container via SSH using the [CLI](../../development/cli/_index.md)
+   by running `platform ssh`.
+2. Change to the `/tmp` directory (or any other non-web-accessible writable directory) with `cd /tmp`.
+3. Download CacheTool with `curl -sLO https://github.com/gordalina/cachetool/releases/latest/download/cachetool.phar`.
+4. Make CacheTool executable with `chmod +x cachetool.phar`.
+5. Check the OPcache status for FastCGI commands.
 
-Note that some applications generate PHP code at runtime based on user configuration.
-If your application does that,
-you can't disable the timestamp validation
-as it would prevent updates to the generated code from being loaded.
+   ```bash
+   php cachetool.phar opcache:status --fcgi=$SOCKET
+   ```
 
-The timestamp validation may be disabled with an `ini` setting,
-and the easiest way to do so is via `.platform.app.yaml`:
+   The `--fcgi=$SOCKET` option ensures the PHP-FPM process on the server connects through the right socket.
+6. Analyze the output to determine the optimal value for `opcache.memory_consumption`.
+   The most important values from CacheTool's output are:
 
-```yaml
+   - `Memory used`
+   - `Memory free`
+   - `Oom restarts` (out of memory restarts)
+     If the value is different than 0, you don't have enough memory allocated to OPcache.
+
+   If `Memory free` is too low or `Oom Restarts` too high,
+   set a higher value for memory consumption.
+7. Set `opcache.memory_consumption`.
+   Note: The unit for `opcache.memory_consumption` is megabytes.
+
+   An example configuration:
+
+   ```yaml {location=".platform.app.yaml"}
+   variables:
+       php:
+           'opcache.memory_consumption': 96
+   ```
+
+8. [Restart PHP-FPM](#restart-php-fpm) and make sure that OPcache works as expected by rerunning CacheTool.
+9. Remove CacheTool by deleting the `cachetools.phar` file.
+
+### Disable OPcache timestamp validation
+
+By default, OPcache checks that the cached version of a file is always up to date.
+This means that every time a cached file is used, OPcache compares it to the file on disk.
+If that file has changed, it gets reloaded and re-cached.
+This allows to support apps that generate compiled PHP code from user configuration.
+
+If you know your code isn't going to change outside of a new deploy,
+you can disable that check and get a small performance improvement.
+
+Timestamp validation can be disabled by adding the following to your [app configuration](../../create-apps/_index.md):
+
+```yaml {location=".platform.app.yaml"}
 variables:
     php:
         'opcache.validate_timestamps': 0
```

Comment:
:question: I just realized the default is `On`. Is `0` really the best to oppose that?